racklog.rkt

```diff
@@ -46,54 +52,53 @@
                       (cond
                         [(null? v) v]
                         [(pair? v) (cons (car v) (lv->list (cdr v)))]
-                        [else (fk 'fail)])))])
+                        [else (fk)])))])
       (if (and (procedure? pred-v) (procedure-arity-includes? pred-v (length args-v)))
-          ((apply pred-v args-v) fk)
-          (fk 'fail)))))
+          (((apply pred-v args-v) sk) fk)
+          (fk)))))
 
 (define (%andmap pred lst . rest)
   (define lsts (cons lst rest))
-  (lambda (fk)
-    (let/racklog-sk sk
+  (lambda (sk)
+    (lambda (fk)
       ; Base case (all lists empty)
-      (let/racklog-fk (fk fk)
-        (sk (foldl (lambda (lst fk) ((%= lst '()) fk)) fk lsts)))
+      (let/racklog-fk fk
+        ((foldr (lambda (lst sk) ((%= lst '()) sk)) sk lsts) fk))
       ; Call and recurse
```

Comment:
It really is 'recur' not cursing over and over again :)

Reply:
Oops! I'll bear that in mind...